esp-hal/src/uart.rs

```diff
@@ -904,8 +870,8 @@ where
     }
 
     /// Read all available bytes from the RX FIFO into the provided buffer and
-    /// returns the number of read bytes. Never blocks
-    pub fn drain_fifo(&mut self, buf: &mut [u8]) -> usize {
+    /// returns the number of read bytes without blocking.
+    pub fn read_bytes(&mut self, buf: &mut [u8]) -> usize {
```

Comment:
Should this be returning `Result<usize, Error>` instead?

Reply:
I don't really see why. Given that we aim to remove `nb`, `read_byte` may end up returning `Option<u8>` and `read_bytes` will not have an error to return.

Reply:
I see, so `read_byte` should actually be returning `nb::Result<(), Infallible>` then.

It just seemed weird to me to drop the error, when the previous `read_bytes` had it and `read_byte` still has it, but if it's going to be cleaned up later then I guess it's fine.